src/api/client/session.rs

```diff
@@ -247,6 +229,74 @@ pub(crate) async fn login_route(
 	})
 }
 
+/// # `POST /_matrix/client/v1/login/get_token`
+///
+/// Allows a logged-in user to get a short-lived token which can be used
+/// to log in with the m.login.token flow.
+///
+/// <https://spec.matrix.org/v1.13/client-server-api/#post_matrixclientv1loginget_token>
+#[tracing::instrument(skip_all, fields(%client), name = "login_token")]
+pub(crate) async fn login_token_route(
+	State(services): State<crate::State>,
+	InsecureClientIp(client): InsecureClientIp,
+	body: Ruma<get_login_token::v1::Request>,
+) -> Result<get_login_token::v1::Response> {
+	if !services.server.config.login_via_existing_session {
+		return Err!(Request(Unknown("Login via an existing session is not enabled")));
+	}
+	// Authentication for this endpoint was made optional, but we need
+	// authentication.
+	let sender_user = body
+		.sender_user
+		.as_ref()
+		.ok_or_else(|| Error::BadRequest(ErrorKind::MissingToken, "Missing access token."))?;
+	let sender_device = body.sender_device.as_ref().expect("user is authenticated");
+
+	// This route SHOULD have UIA
+	// TODO: How do we make only UIA sessions that have not been used before valid?
```

Comment:
If this is supposed to be a requirement on all endpoints with UIA, then I don't believe this is done (inherited by upstream) anywhere and probably should make a general issue for closing this mild UIA security gap. Creating a new CF for storing this type of information might be necessary.

Regardless if this endpoint doesn't have any inherent requirement for it, then I won't make it a blocker.

Reply:
This is specific to this endpoint, to prevent the case where an evil client uses their single UIA to generate an arbitrary amount of sessions. 
It's a bit pointless in our case at the moment as the only UIA a user is likely to use is password, which can be seen and reused by the client directly anyway.

Reply:
In that case, it's really a suggestion for improved security rather than a "must" or requirement and this mild security gap can be fixed another time.